### PR TITLE
[Improvement] enhance show data types stmt

### PR DESCRIPTION
**Pull request by ZashJie**
enhance show data types stmt
mysql> show data types;
+----------------+------+
| TypeName       | Size |
+----------------+------+
| ARRAY          | 32   |
| BIGINT         | 8    |
| BITMAP         | 16   |
| BOOLEAN        | 1    |
| CHAR           | 16   |
| DATE           | 16   |
| DATETIME       | 16   |
| DATETIMEV2     | 8    |
| DATEV2         | 4    |
| DECIMAL128     | 16   |
| DECIMAL32      | 4    |
| DECIMAL64      | 8    |
| DECIMALV2      | 16   |
| DOUBLE         | 8    |
| FLOAT          | 4    |
| HLL            | 16   |
| INT            | 4    |
| JSONB          | 16   |
| LARGEINT       | 16   |
| MAP            | 24   |
| QUANTILE_STATE | 16   |
| SMALLINT       | 2    |
| STRING         | 16   |
| TINYINT        | 1    |
| VARCHAR        | 16   |
+----------------+------+
25 rows in set (0.01 sec)

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowDataTypesStmt.java [hunk 1]
```diff
@@ -41,6 +43,33 @@ public static ArrayList<PrimitiveType> getTypes() {
...
 
+    public static List<List<String>> getFilterTypes() {
+        ArrayList<PrimitiveType> supportedTypes = getTypes();
+        List<List<String>> rows = Lists.newArrayList();
+        boolean hasDecimalv3 = false;
+        for (PrimitiveType type : supportedTypes) {
+            List<String> row = new ArrayList<>();
+            if (type.isAvailableInDdl()) {
+                if (type.isDecimalV2Type()) {
+                    row.add("DECIMAL");
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowDataTypesStmt.java:54]: We should not handle type name here, should be handled in PrimitiveType too.

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.28 seconds
 stream load tsv:          427 seconds loaded 74807831229 Bytes, about 167 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230426023439_clickbench_pr_135260.html

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run p1

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp1

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp1

**Comment by ZashJie** [comment]
run buildp0p1

**Comment by ZashJie** [comment]
run buildp1

**Comment by ZashJie** [comment]
run buildp1

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildp1

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildp0

**Comment by dataroaring** [comment]
run buildall

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run p0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildall

**Comment by ZashJie** [comment]
run p0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildp0

**Comment by ZashJie** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**